lib/active_shipping/location.rb

```diff
@@ -42,48 +60,50 @@ def initialize(options = {})
     end
 
     def self.from(object, options = {})
-      return object if object.is_a? ActiveShipping::Location
-      attr_mappings = {
-        :name => [:name],
-        :country => [:country_code, :country],
-        :postal_code => [:postal_code, :zip, :postal],
-        :province => [:province_code, :state_code, :territory_code, :region_code, :province, :state, :territory, :region],
-        :city => [:city, :town],
-        :address1 => [:address1, :address, :street],
-        :address2 => [:address2],
-        :address3 => [:address3],
-        :phone => [:phone, :phone_number],
-        :fax => [:fax, :fax_number],
-        :address_type => [:address_type],
-        :company_name => [:company, :company_name]
-      }
+      return object if object.is_a?(ActiveShipping::Location)
+
       attributes = {}
+
       hash_access = begin
         object[:some_symbol]
         true
       rescue
         false
       end
-      attr_mappings.each do |pair|
+
+      ATTRIBUTE_MAPPINGS.each do |pair|
```

Comment:
Yeah, mentioned it offline but `object.send(sym) if object.respond_to?(sym)` should be synonymous to `object.try(sym)` in this case.